content/en/blog/gitops-your-falco-rules/index.md

```diff
@@ -0,0 +1,805 @@
+    falcoVersions: http://localhost:8765/versions
+    rulesfilesdir: /etc/falco/rules.d
+    refs:
+      # Leave out the predefined ruleset for the moment
+      # - falco-rules:0
+      - ghcr.io/vjjmiras/cool-falco-rules/custom-rules:main
+indexes:
+  - name: falcosecurity
+    url: https://falcosecurity.github.io/falcoctl/index.yaml
+```
+
+To verify the `custom_falco_rules.yaml` file doesn't exist, we'll
+```
+$ stat /etc/falco/rules.d/custom_falco_rules.yaml
+stat: cannot access '/etc/falco/rules.d/custom_falco_rules.yaml': No such file or directory
+
+$ sudo systemctl restart falcoctl-artifact-follow
+$ sudo systemctl status falcoctl-artifact-follow
+$ sudo cat /etc/falco/rules.d/custom_falco_rules.yaml
+
+- list: falco_binaries
+  items: [falcoctl]
+```
+
+Before we move on, observe the alert that Falco has been giving when the `falcoctl-artifact-follow` service restarts:
+
+```!
+$ sudo systemctl status falco
+
+...
+Apr 14 15:31:19 vagrant falco[24944]: 15:31:19.049935845: Error File below /etc opened for writing (user=root user_loginuid=-1 command=falcoctl artifact follow --allowed-types=rulesfile pid=24990 parent=falcoctl pcmdline=falcoctl artifact follow --allowed-types=rulesfile file=/etc/falco/._check_writable program=falcoctl gparent=<NA> ggparent=<NA> gggparent=<NA> container_id=host image=<NA>)
+...
+```
+
+We can get rid of that alert by extending our `custom-falco-rules.yaml` ruleset file, generate a new OCI artifact and let `falcoctl` download it. It'll be the perfect example to test our new configuration.
+
+The following macro adds a new condition not to trigger the rule. The condition is that the process trying to write in the the `/etc/` directory is `falcoctl`. We added that list previously to have some content in the file.
+
+```
+- list: falco_binaries
+  items: [falcoctl]
+
+- macro: write_etc_common
+  append: true
+  condition: and not proc.pname in (falco_binaries)
+```
+
+```
+$ git commit -a -s -m 'new: adding the macro to the list'
+
+[main b82c5e6] adding the macro to the list
+ 1 file changed, 4 insertions(+)
+
+$ git push
+
+Enumerating objects: 5, done.
+Counting objects: 100% (5/5), done.
+Delta compression using up to 12 threads
+Compressing objects: 100% (3/3), done.
+Writing objects: 100% (3/3), 439 bytes | 219.00 KiB/s, done.
+Total 3 (delta 0), reused 0 (delta 0), pack-reused 0
+To https://github.com/vjjmiras/cool-falco-rules.git
+   957bbcc..b82c5e6  main -> main
+ ```
+ 
+This should trigger the GitHub actions workflow and in a few minutes, we should have an updated ruleset file in our `/etc/falco` directory:
+ 
+```
+$ sudo cat /etc/falco/rules.d/custom_falco_rules.yaml
+
+- list: falco_binaries
+  items: [falcoctl]
+
+- macro: write_etc_common
+  append: true
+  condition: and not proc.pname in (falco_binaries)
+ ```
+ 
+ `falcoctl` also takes care of telling `falco` that a new rules file is present. If you look at the status of the `falco` service you should see the following line:
```

Comment:
This behavior is automatic, Falco detects by itself when a rules file has been modified. It checks it then SIGHUP to reload